dropshot/src/handler.rs

```diff
@@ -584,6 +669,29 @@ impl HttpResponse for Response<Body> {
     }
 }
 
+impl<T, E> HttpResponse for Result<T, E>
```

Comment:
I know you've already talked about this above, but this is such an important impl that perhaps it warrants a comment?

Implement HttpResponse for any Result<HttpResponse, HttpResponseError> to address the most common use case

Now that I've written that... meh... I don't know. Do what you like!